sdk/python/feast/repo_config.py

```diff
@@ -32,13 +34,29 @@ class Config:
         extra = "allow"
 
 
-class FeastConfigBaseModel(BaseModel):
+class FeastOfflineStoreConfigBaseModel(BaseModel):
     """ Feast Pydantic Configuration Class """
 
     class Config:
         arbitrary_types_allowed = True
         extra = "forbid"
 
+    @abstractmethod
+    def supports_data_source(self, data_source: DataSource) -> bool:
+        ...
+
+
+class FeastOnlineStoreConfigBaseModel(BaseModel):
+    """ Feast Pydantic Configuration Class """
+
+    class Config:
+        arbitrary_types_allowed = True
+        extra = "forbid"
+
+    @abstractmethod
+    def supports_offline_store(self, offline_store: Any) -> bool:
+        ...
+
```

Comment:
Why does online store need to define which offline stores to support? The online store and offline store do no interact with each other directly.

I'd say let providers define which online store & offline store they support.

Reply:
> Why does online store need to define which offline stores to support? The online store and offline store do no interact with each other directly.
> 
> I'd say let providers define which online store & offline store they support.

Spoke offline - I'm going to scrap this PR because there's no explicitly unsupported combination at the moment. We should wait until the need for this kind of API emerges, possibly after we have an AWS provider.